Test failures on 32-bit architectures

Hi @marcelm 

Thanks for your work on xopen!

The `test_open_many_writers[.xz]` is failing 32-bit architectures with a `MemoryError`
Is this actually not compatible with 32-bit architectures?

The full log can be found [here](https://ci.debian.net/data/autopkgtest/testing/i386/p/python-xopen/15623493/log.gz) and I am also pasting the relevant part below. Please consider to fix this, and let me know if this is intended in any case.

```
=================================== FAILURES ===================================
_________________________ test_open_many_writers[.xz] __________________________

tmp_path = PosixPath('/tmp/pytest-of-debci/pytest-0/test_open_many_writers__xz_0')
ext = '.xz'

    def test_open_many_writers(tmp_path, ext):
        files = []
        for i in range(1, 61):
            path = tmp_path / f"{i:03d}.txt.{ext}"
>           f = xopen(path, "wb", threads=2)

tests/test_xopen.py:635: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3/dist-packages/xopen/__init__.py:737: in xopen
    return _open_xz(filename, mode)
/usr/lib/python3/dist-packages/xopen/__init__.py:581: in _open_xz
    return lzma.open(filename, mode)
/usr/lib/python3.9/lzma.py:309: in open
    binary_file = LZMAFile(filename, lz_mode, format=format, check=check,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <lzma.LZMAFile object at 0xf6bf90b8>
filename = '/tmp/pytest-of-debci/pytest-0/test_open_many_writers__xz_0/044.txt..xz'
mode = 'wb'

    def __init__(self, filename=None, mode="r", *,
                 format=None, check=-1, preset=None, filters=None):
        """Open an LZMA-compressed file in binary mode.
    
        filename can be either an actual file name (given as a str,
        bytes, or PathLike object), in which case the named file is
        opened, or it can be an existing file object to read from or
        write to.
    
        mode can be "r" for reading (default), "w" for (over)writing,
        "x" for creating exclusively, or "a" for appending. These can
        equivalently be given as "rb", "wb", "xb" and "ab" respectively.
    
        format specifies the container format to use for the file.
        If mode is "r", this defaults to FORMAT_AUTO. Otherwise, the
        default is FORMAT_XZ.
    
        check specifies the integrity check to use. This argument can
        only be used when opening a file for writing. For FORMAT_XZ,
        the default is CHECK_CRC64. FORMAT_ALONE and FORMAT_RAW do not
        support integrity checks - for these formats, check must be
        omitted, or be CHECK_NONE.
    
        When opening a file for reading, the *preset* argument is not
        meaningful, and should be omitted. The *filters* argument should
        also be omitted, except when format is FORMAT_RAW (in which case
        it is required).
    
        When opening a file for writing, the settings used by the
        compressor can be specified either as a preset compression
        level (with the *preset* argument), or in detail as a custom
        filter chain (with the *filters* argument). For FORMAT_XZ and
        FORMAT_ALONE, the default is to use the PRESET_DEFAULT preset
        level. For FORMAT_RAW, the caller must always specify a filter
        chain; the raw compressor does not support preset compression
        levels.
    
        preset (if provided) should be an integer in the range 0-9,
        optionally OR-ed with the constant PRESET_EXTREME.
    
        filters (if provided) should be a sequence of dicts. Each dict
        should have an entry for "id" indicating ID of the filter, plus
        additional entries for options to the filter.
        """
        self._fp = None
        self._closefp = False
        self._mode = _MODE_CLOSED
    
        if mode in ("r", "rb"):
            if check != -1:
                raise ValueError("Cannot specify an integrity check "
                                 "when opening a file for reading")
            if preset is not None:
                raise ValueError("Cannot specify a preset compression "
                                 "level when opening a file for reading")
            if format is None:
                format = FORMAT_AUTO
            mode_code = _MODE_READ
        elif mode in ("w", "wb", "a", "ab", "x", "xb"):
            if format is None:
                format = FORMAT_XZ
            mode_code = _MODE_WRITE
>           self._compressor = LZMACompressor(format=format, check=check,
                                              preset=preset, filters=filters)
E           MemoryError

/usr/lib/python3.9/lzma.py:111: MemoryError
```

--- a/tests/test_xopen.py
+++ b/tests/test_xopen.py
@@ -630,8 +630,11 @@ def test_xopen_falls_back_to_bzip2_open(lacking_pbzip2_permissions):
 
 def test_open_many_writers(tmp_path, ext):
     files = []
-    for i in range(1, 61):
-        path = tmp_path / f"{i:03d}.txt.{ext}"
+    # Because lzma.open allocates a lot of memory,
+    # open fewer files to avoid MemoryError on 32-bit architectures
+    n = 21 if ext == ".xz" else 61
+    for i in range(1, n):
+        path = tmp_path / f"{i:03d}.txt{ext}"
         f = xopen(path, "wb", threads=2)
         f.write(b"hello")
         files.append(f)

